src/test/java/api/RelatedInstanceTest.java

```diff
@@ -0,0 +1,308 @@
+package api;
+
+import static api.support.InstanceSamples.nod;
+import static io.vertx.core.json.JsonObject.mapFrom;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.folio.inventory.domain.instances.Instance;
+import org.folio.inventory.domain.instances.RelatedInstance;
+import org.junit.Test;
+
+import api.support.ApiTests;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+public class RelatedInstanceTest extends ApiTests {
+
+  @Test
+  public void testRelatedInstance() {
+    String id = UUID.randomUUID().toString();
+    String instanceId = UUID.randomUUID().toString();
+    String relatedInstanceId = UUID.randomUUID().toString();
+    String relatedInstanceTypeId = UUID.randomUUID().toString();
+    RelatedInstance relatedInstance = new RelatedInstance(
+        id,
+        instanceId,
+        relatedInstanceId,
+        relatedInstanceTypeId);
+
+    JsonObject relatedInstanceJson = mapFrom(relatedInstance);
+
+    RelatedInstance sameRelatedInstance = RelatedInstance.from(relatedInstanceJson, instanceId);
+
+    assertEquals(id, sameRelatedInstance.id);
+    assertEquals(instanceId, sameRelatedInstance.instanceId);
+    assertEquals(relatedInstanceId, sameRelatedInstance.relatedInstanceId);
+    assertEquals(relatedInstanceTypeId, sameRelatedInstance.relatedInstanceTypeId);
+
+    assertEquals(relatedInstance.id, sameRelatedInstance.id);
+    assertEquals(relatedInstance.instanceId, sameRelatedInstance.instanceId);
+    assertEquals(relatedInstance.relatedInstanceId, sameRelatedInstance.relatedInstanceId);
+    assertEquals(relatedInstance.relatedInstanceTypeId, sameRelatedInstance.relatedInstanceTypeId);
+
+    RelatedInstance inverseRelatedInstance = RelatedInstance.from(relatedInstanceJson, relatedInstanceId);
+
+    assertEquals(id, inverseRelatedInstance.id);
+    assertEquals(relatedInstanceId, inverseRelatedInstance.instanceId);
+    assertEquals(instanceId, inverseRelatedInstance.relatedInstanceId);
+    assertEquals(relatedInstanceTypeId, inverseRelatedInstance.relatedInstanceTypeId);
+
+    assertEquals(relatedInstance.id, inverseRelatedInstance.id);
+    assertEquals(relatedInstance.instanceId, inverseRelatedInstance.relatedInstanceId);
+    assertEquals(relatedInstance.relatedInstanceId, inverseRelatedInstance.instanceId);
+    assertEquals(relatedInstance.relatedInstanceTypeId, inverseRelatedInstance.relatedInstanceTypeId);
+  }
+
+  @Test
+  public void canCreateInstanceWithRelatedInstances() throws Exception {
```

Comment:
Personally, I would use the sneaky throws trait rather than indicate this method throws a general exception (which provides almost no information to the reader)

Ideally, the sneaky throws trait could be put on only the methods invoked in the test that throw checked exceptions. That way other tests can benefit from this as well.

```suggestion
  @SneakyThrows
  public void canCreateInstanceWithRelatedInstances() {
```

Reply:
Nice. Learn something new every day. Thanks.